src/meteodatalab/operators/regrid.py

```diff
@@ -404,3 +407,77 @@ def icon2rotlatlon(field: xr.DataArray) -> xr.DataArray:
     )
 
     return _icon2regular(field, dst, indices, weights)
+
+
+def _linear_weights(xy, uv):
+    """Compute indices and weights for barycentric linear interpolation."""
+    tri = Delaunay(xy)
+    simplex = tri.find_simplex(uv)
+    isfound = simplex != -1
+    vertices = np.take(tri.simplices, simplex, axis=0)
+    indices = np.where(isfound[:, None], vertices, 0)
+
+    # note that zero is a valid index
+    # however, a full line of zeros is interpreted as out of bounds
+    # in the interpolation step
+
+    temp = np.take(tri.transform, simplex, axis=0)
+    delta = uv - temp[:, 2]
+    bary = np.einsum("njk,nk->nj", temp[:, :2, :], delta)
+    wgts = np.hstack((bary, 1 - bary.sum(axis=1, keepdims=True)))
+    weights = np.where(isfound[:, None], wgts, 0)
+
+    return indices, weights
+
+
+def _cropped_domain(xy, uv, buffer=1000):
+    """Crop the grid to output domain."""
+    xmin, ymin = np.min(uv, axis=0) - buffer
+    xmax, ymax = np.max(uv, axis=0) + buffer
+    x, y = xy.T
+    mask = (xmin < x) & (x < xmax) & (ymin < y) & (y < ymax)
+    [idx] = np.nonzero(mask)
+    indices, weights = _linear_weights(xy[mask], uv)
+    return idx[indices], weights
+
+
+def iconremap(
```

Comment:
maybe a comment on the API: 
we have 
- icon2geolatlon: implicit method (RBF), dst grid can not be passed
- icon2rotlatlon: implicit method (RBF), dst grid can not be passed
- iconremap: method passed as argument (byc), dst grid passed by argument

could we homogenise the api, maybe also 1 single entry point that dispatches according to the method?

Reply:
I don't think that multiplexing the function through a single interface is a good design. Especially in this case where the behavior change between the methods is so large. On the one hand, the weights are pre computed so the destination grid is dictated by that and on the other the weights are computed on the fly so a destination grid can be defined.

Reply:
If you are not considering refactoring the api, I would at least make it explicit that pre-computed weights from RBF are used, either through docstrings or by renaming those methods.